.github/actions/wait-for-es-data/action.yml

```diff
@@ -0,0 +1,35 @@
+name: "Wait for ES Data"
+description: "Waits for data on ES indices"
+inputs:
+  index:
+    description: "Which ES index to look at"
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Wait for data
+      env:
+        ES_HOST: http://localhost:9200
+      shell: bash
+      run: |
+        URL="${ES_HOST}/${{ inputs.index }}/_search?size=0"
+        HEADER="Content-Type: application/json"
+        MAX_RETRIES=100
+
+        echo "Checking index: ${{ inputs.index }}"
+
+        for ((i = 1; i <= MAX_RETRIES; i++)); do
+            result=$(curl -s -X GET "$URL" -H "$HEADER" | jq '.hits.total.value')
+
+            if [ "$result" -gt 0 ]; then
+                echo "Data found: $result documents in index ${{ inputs.index }}"
+                exit 0
+            fi
+
+            echo "No data yet. Retrying in 1s... ($i/$MAX_RETRIES)"
+            sleep 1
+        done
```

Comment:
The waiting loop is currently based on a hardcoded maximum retries count of 100. Since this action is intended for reuse, consider introducing a max_retries input parameter with a default value of 100, along with an explanation of how the retries work, as the timeout may vary depending on the connection to different providers.

Alternatively, you could rewrite the loop to use a time-based approach, specifying the total wait time in seconds rather than relying on the maximum retries.

Reply:
added `retries` input. because each retry is 1s, this is effectively time based (100 retries is roughly 100s, plus the time it takes to execute the curl command). 

note that the whole point of using this action is to avoid tweaking sleep period time for different consumers, so they can all just "wait for data". some may take 20 seconds, some 24, 26, 30 or 40. if it takes more than 100 seconds, something is probably not working.